pkg/tfbridge/auto_aliasing.go

```diff
@@ -125,409 +282,444 @@ func (info *ProviderInfo) ApplyAutoAliases() error {
 			}
 		}
 	}
+	return currentVersion, nil
+}
 
-	rMap := info.P.ResourcesMap()
-	dMap := info.P.DataSourcesMap()
+func (autoAliasHelper) computeMaxItemsOneOverrides(
+	schemaMap shim.SchemaMap,
+	schemaInfos map[string]*SchemaInfo,
+	history map[string]*fieldHistory,
+) maxItemsOneOverrides {
+	if schemaMap == nil {
+		return nil
+	}
+	over := make(maxItemsOneOverrides)
+	fh := &fieldHistory{Fields: history}
+	walk.VisitSchemaMap(schemaMap, func(sp walk.SchemaPath, s shim.Schema) {
+		info := LookupSchemaInfoMapPath(sp, schemaInfos)
+
+		// If the user already set a MaxItemsOne override, respect it.
+		if info != nil && info.MaxItemsOne != nil {
+			return
```

Comment:
As I understand it, this means that if I remove a manual `MaxItemsOne` override, the next time we run `make tfgen` we will observe a change?

I believe that this is a divergence from existing behavior.

Reply:
Hmm. Really appreciate this kind of analysis! Let me add tests to make explicit what happens here.

My intention here was this. If the user set in MaxItemsOne in current version, we should ever be changing that by automatic aliasing. That's what the code says. Note that maxItemsOneOverrides  are not saved to history. They're only for the current version, for moving data from build-time to runtime.

If you set MaxItemsOne=true in V=1.0.0 and you're building V=1.0.1 and you removed MaxItemsOne  in V=1.0.1, then what's happening here I believe is that effective MaxItemsOne=true will be loaded from aliasHistory, so `prev.MaxItemsOne == &true`, this code will compare it to actual, and if actual=false it will decide to put the override back to match the previous value. So presumably the schema of the provider will not change.

Reply:
Great. It sounds like your describing the behavior we want:
- Overrides in `resources.go` are respected.
- They remain sticky until the next major version.